plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -378,6 +414,41 @@ func (r *ReferenceManager) Admit(ctx context.Context, a admission.Attributes, o
 				})
 			}
 		}
+	case core.Kind("BackupBucket"):
+		if operation == admission.Delete {
+			// The "delete endpoint" handler of the k8s.io/apiserver library calls the admission controllers
+			// handling DELETE requests with empty objects:
+			// https://github.com/kubernetes/apiserver/blob/release-1.25/pkg/endpoints/handlers/delete.go#L126
+			// Consequently, a.GetObject() will be nil.
+			err = r.validateBackupBucketDeletion(ctx, a.GetName())
```

Comment:
IIRC when `DeleteCollection` is used, the `a.GetName()` function will return an empty name. You can check https://github.com/gardener/gardener/blob/0fea80282ee82aadc3f5b081714bf5741b10f17e/plugin/pkg/global/deletionconfirmation/admission.go#L239-L245 and whether this still holds true.

Reply:
I thought about that too, should we move this check to `DeletionConfirmation`? There we already have means to deal with this.

Reply:
I would rather keep `DeletionConfirmation` plugin only for the `confirmation.gardener.cloud/deletion` annotation. For instance there is handling for `DeleteCollection` also in the managedseed plugins: https://github.com/gardener/gardener/blob/a2eebdf4ff99f8186184ee6de685586db8e1b37b/plugin/pkg/shoot/managedseed/admission.go#L132 and https://github.com/gardener/gardener/blob/a2eebdf4ff99f8186184ee6de685586db8e1b37b/plugin/pkg/managedseed/shoot/admission.go#L132. 
@rfranzke wdyt?

Reply:
Addressed in https://github.com/gardener/gardener/pull/7065/commits/64d976cc595cc3a159893fd07878014b11f5c377